### PR TITLE
make emergency lights (de)constructable

### PR DESCRIPTION
**Pull request by Boolean-Buckeye**
* add an emergency light node to the LightFixture construction graph
* add a recipe entry to the construction menu for emergency light
* make emergency light entity at the emergency light node on the graph

## About the PR

I made emergency lights constructible and deconstructible.

## Why / Balance

When I was playing the game, I right clicked an emergency light and there was an option to "Begin Deconstructing". which I clicked, and then I examined the emergency light and it said to use a screwdriver next. But when I went to do that, nothing happened. It turns out this examine shows up I think because the parent of the emergency light entity has it at the light tube point on the light fixture construction graph by default. So, being annoyed, I went and looked up how to code for ss14 to fix this. The recipe is one steel and one glass, and I figure since the emergency light looks almost exactly like the empty light fixture, just adding glass to the recipe would be fine. It's a pretty simple (and very small) looking structure with not much use so I think it would be overkill to require using a lathe to print an emergency light like with the other wall lights like bulbs and tubes. 

## Technical details

I fixed it by adding a new node on the light fixture graph for emergency lights. This only required changing the yamls to:
* add a construction node for emergency lights in light fixture graph
* add the recipe to the construction menu
* emergency light entity construction component set to this new node

## Requirements

- [x ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x ] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl:
- add: Emergency lights can be constructed and deconstructed

### CODE REVIEW

**Comment by slarticodefast** [comment]
The test fail is real.
EmergencyLight inherits this from AlwaysPoweredWallLight, meaning if you destroy it by force, you will get 2 steel sheets, while constructing it only needs 1.
grafik

**Comment by Boolean-Buckeye** [comment]
> The test fail is real. EmergencyLight inherits this from AlwaysPoweredWallLight, meaning if you destroy it by force, you will get 2 steel sheets, while constructing it only needs 1. grafik

Hi, thanks for the explanation, I am impressed by this testing, this is my first time trying to do coding , I updated the PR so that the emergency light is destroyed into 1 steel and 1 glass shard

**Comment by Cojoke-dot** [comment]
mmmm... it says they have an internal battery, maybe they should need a battery?

**Comment by slarticodefast** [comment]
> mmmm... it says they have an internal battery, maybe they should need a battery?

All the other lights are really cheap to build as well, so I think it's fine without for consistency reasons.

**Comment by deltanedas** [comment]
if anything it should be a capacitor not battery, that would be really annoying

**Comment by Boolean-Buckeye** [comment]
@slarticodefast is it still awaiting changes ?

File: Resources/Prototypes/Recipes/Construction/structures.yml [hunk 1]
```diff
@@ -1456,6 +1456,25 @@
...
+  icon:
+    sprite: Structures/Wallmounts/Lighting/emergency_light.rsi
+    state: base
+  objectType: Structure
+  placementMode: SnapgridCenter
+  canRotate: true
+  canBuildInImpassable: false
+  conditions:
+  # Same here. - 20kdc
+    - !type:TileNotBlocked
```
Comment (deathride58) [inline Resources/Prototypes/Recipes/Construction/structures.yml:1476]: Comment is an unnecessary copypaste
```suggestion
  conditions:
    - !type:TileNotBlocked
```

**Review by deathride58 — CHANGES_REQUESTED** [review]
PR looks good, but there's one little nitpick

**Review by deathride58 — APPROVED** [review]

**Comment by chromiumboy** [comment]
Thank you for your first contribution!

The code looks good and it all tested fine in-game

**Pull request merged**